emukit/quadrature/methods/vanilla_bq.py

```diff
@@ -52,18 +53,55 @@ def predict_base_with_full_covariance(self, X_pred: np.ndarray) -> Tuple[np.ndar
         m, cov = self.base_gp.predict_with_full_covariance(X_pred)
         return m, cov, m, cov
 
-    def integrate(self) -> Tuple[float, float]:
+    def integrate(self, measure: UniformMeasure = None) -> Tuple[float, float]:
         """
         Computes an estimator of the integral as well as its variance.
 
+        :param measure: The measure which is integrated against (default is Lebesgue)
         :returns: estimator of integral and its variance
         """
-        integral_mean, kernel_mean_X = self._compute_integral_mean_and_kernel_mean()
+        if measure is None:
+            integral_mean, integral_var = self._integrate_lebesgue()
+        elif isinstance(measure, UniformMeasure):
+            integral_mean, integral_var = self._integrate_uniform(measure)
+        else:
+            raise ValueError('unknown measure')
+        return integral_mean, integral_var
+
+    def _integrate_lebesgue(self) -> Tuple[float, float]:
+        """
+        Computes integral against Lebesgue measure
+        :returns: estimator of integral and its variance
+        """
+        kernel_mean_X = self.base_gp.kern.qK(self.X)
+        integral_mean = np.dot(kernel_mean_X, self.base_gp.graminv_residual())[0, 0]
         integral_var = self.base_gp.kern.qKq() - np.square(lapack.dtrtrs(self.base_gp.gram_chol(), kernel_mean_X.T,
                                                            lower=1)[0]).sum(axis=0, keepdims=True)[0][0]
         return integral_mean, integral_var
 
-    # helpers
+    def _integrate_uniform(self, measure: UniformMeasure) -> Tuple[float, float]:
+        """
+        Computes integral against Uniform measure.
+        :param measure: A uniform measure
+        :returns: estimator of integral and its variance
+        """
+        # get max of both lower bounds and min of both upper bounds and integrate over the resulting bounds.
+        # This is equivalent (up to the uniform density) to integrating with respect to the uniform measure.
+        integral_bounds = self.integral_bounds.bounds
+        uniform_bounds = measure.bounds
+        old_integral_bound_list = self.integral_bounds.bounds.copy()
+        new_integral_bound_list = [(max(int_bounds[0], uni_bounds[0]), min(int_bounds[1], uni_bounds[1]))
+                                   for int_bounds, uni_bounds in zip(integral_bounds, uniform_bounds)]
+
+        # setting new bounds also checks bound validity
+        self.integral_bounds = new_integral_bound_list
+        integral_mean_lebesgue, integral_var_lebesgue = self._integrate_lebesgue()
```

Comment:
The other approach would be to input `integration_bounds` to `self._integrate_lebesgue()` and then into all kernel methods that require the bounds instead of having the bounds as an attribute on the kernel. This would avoid needing to worry about the integration bounds setting and resetting. Thoughts?

Reply:
I see your point, but I think that might cause bigger problems because then you need to make sure that whoever designs any integration method use the same bounds for all the methods of the kernel, which is very error prone. Having the attribute in the kernel (the only place where integration happens) avoids that. I'd rather make sure to write a nice setter.

Also, to be clear, I do not want the user to change the integral bounds. They are the parameter space and I see no point in changing them. It is just a way to internally do the uniform measure.